feat(cli): Add autocompletion support

BonnetAdam opened the pull request:
feat(cli): add autocompletion support for stack names in various commands
feat(cli): add autocompletion support for database dump and restore commands
feat(cli): add autocompletion support for stack service names in various commands
feat(cli): add autocompletion support for yml file names in various commands
fix(constant.py): add list of supported file extensions for autocompletion
fix(services/base.py): add check for name attribute in BaseService to prevent missing attribute error
feat(services/base.py): add list method to BaseService to get a list of all services or only their names
feat(utils/autocomplete.py): add autocompletion functions for stack names, yml file names, and stack service names

BonnetAdam committed:
feat(cli): add autocompletion support for stack names in various commands
feat(cli): add autocompletion support for database dump and restore commands
feat(cli): add autocompletion support for stack service names in various commands
feat(cli): add autocompletion support for yml file names in various commands
fix(constant.py): add list of supported file extensions for autocompletion
fix(services/base.py): add check for name attribute in BaseService to prevent missing attribute error
feat(services/base.py): add list method to BaseService to get a list of all services or only their names
feat(utils/autocomplete.py): add autocompletion functions for stack names, yml file names, and stack service names

remyz17 commented on odooghost/utils/autocomplete.py:
```diff
@@ -0,0 +1,26 @@
+            yield stack.name
+
+
+def ac_yml_files(incomplete: str) -> Generator:
```
function name should be renamed

remyz17 commented on odooghost/services/base.py:
```diff
@@ -20,10 +20,14 @@
+    def __init__(self, stack_config: "StackConfig") -> None:
         self.stack_config = stack_config
         self.stack_name = stack_config.name
+        self._check_name_attribute()
```
why this ? why removing name from service subclasss constructor ?

BonnetAdam replied:
I did that to be able to use the function ac_stacks_services with BaseService.list...

remyz17 commented on odooghost/constant.py:
```diff
@@ -17,6 +17,8 @@
 IS_DARWIN_PLARFORM = sys.platform == "darwin"
 IS_LINUX_PLATFORM = sys.platform == "linux"
 
+EXTENSIONS: list = ["yml", "yaml", "json"]
```
should be renamed to a more meaning name

BonnetAdam committed:
refactor(cli): rename ac_yml_files function to ac_stack_configs for better semantics and clarity
refactor(cli): update references to ac_yml_files to ac_stack_configs in config.py and root.py files to reflect the function name change
refactor(constant): rename EXTENSIONS constant to STACK_CONFIG_FILE_EXTENSIONS for better semantics and clarity

remyz17 commented on odooghost/services/base.py:
```diff
@@ -20,10 +20,14 @@
+
+    def _check_name_attribute(self):
+        if not self.name:
+            raise exceptions.ServiceAttributeMissing("The attribute name is required")
```
RuntimeError is preferred here, because this should never happen and we don't want it to trigger a StackException as it's more like an internal problem not related to any stack

BonnetAdam committed:
refactor(exceptions.py): remove unused ServiceException and ServiceAttributeMissing classes
fix(base.py): replace ServiceAttributeMissing exception with a RuntimeError to improve error handling and messaging
